win32ss/gdi/gdi32/objects/bitmap.c

```diff
@@ -731,6 +731,8 @@ SetDIBitsToDevice(
 
     cjBmpScanSize = DIB_BitmapMaxBitsSize((LPBITMAPINFO) lpbmi, ScanLines);
 
+    ((PW32CLIENTINFO)NtCurrentTeb()->Win32ClientInfo)->cSpins = 0;
```

Comment:
Maybe do it just before `GdiGetHandleUserData` or the NtGdi call? (same question for some other places.)

Reply:
Tracing reveals that Windows seems to place it just before the RtlAllocateHeap calls that write to pvSafeBits, so if anything, I would judge the original developers.